packages/git/src/browser/diff/git-diff-widget.tsx

```diff
@@ -138,15 +141,11 @@ export class GitDiffWidget extends GitNavigableListWidget<GitFileChangeNode> imp
         return <div className='diff-header'>{...children}</div>;
     }
 
-    protected renderHeaderRow({ name, value, classNames }: { name: string, value: React.ReactNode, classNames?: string[] }): React.ReactNode {
-        if (value === null) {
-            return null;
+    protected renderRepositoryHeader(): React.ReactNode {
+        if (this.options && this.options.uri) {
+            return this.renderHeaderRow({ name: 'repository', value: this.getRepositoryLabel(this.options.uri) });
         }
-        const className = ['header-row', ...(classNames || [])].join(' ');
-        return <div key={name} className={className}>
-            <div className='theia-header'>{name}</div>
-            <div className='header-value'>{value}</div>
-        </div>;
+        return undefined;
```

Comment:
AFAIK we once decided to return `undefined`.

Reply:
https://github.com/theia-ide/theia/blob/100c14177b71d54d5585f3af20e6a35fe33d648c/packages/core/src/browser/tree/tree-widget.tsx#L291

https://github.com/theia-ide/theia/blob/100c14177b71d54d5585f3af20e6a35fe33d648c/packages/core/src/browser/tree/tree-widget.tsx#L305

https://github.com/theia-ide/theia/blob/100c14177b71d54d5585f3af20e6a35fe33d648c/packages/core/src/browser/tree/tree-widget.tsx#L321

But

https://github.com/theia-ide/theia/blob/100c14177b71d54d5585f3af20e6a35fe33d648c/packages/core/src/browser/tree/tree-widget.tsx#L524

What is the policy then? +1 for anything, consistent. CC: @akosyakov @AlexTugarev

Reply:
Coding guidelines in theia: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#null-and-undefined

Reply:
agree, it would be good to align. I've seen `return undefined` mostly in such cases.

Reply:
Please open a follow-up. We always used `null` before switching from Phosphor to React. If the policy has changed, we have to update the code and notify others.

Reply:
The policy did not change. At least if the coding guidelines in theia wiki are such a policy for us. The rule to use `undefined` instead of `null` (what I've linked above) is in it since the initial draft from 25th of feb.2017. 
Anyways, I will open a follow-up.

Reply:
https://github.com/theia-ide/theia/issues/3181